<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TINC introduction â€¢ TINC</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TINC introduction">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TINC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/TINC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/TINC">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>TINC introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/TINC/blob/master/vignettes/TINC.Rmd"><code>vignettes/TINC.Rmd</code></a></small>
      <div class="hidden name"><code>TINC.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(TINC)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="co">#&gt;  [ BMix - Binomial and Beta-Binomial mixture models ] </span></a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="co">#&gt; GitHub :  caravagn/BMix </span></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co">#&gt;    WWW :  https://caravagn.github.io/BMix/ </span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-7" data-line-number="7"><span class="co">#&gt;  &gt; BMix is part of the "evoverse" [https://bit.ly/2orn94e] - a collection of packages to implement Cancer Evolution analyses from cancer sequencing data.</span></a>
<a class="sourceLine" id="cb1-8" data-line-number="8"><span class="co">#&gt; Warning: replacing previous import 'Biostrings::translate' by</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9"><span class="co">#&gt; 'seqinr::translate' when loading 'dndscv'</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11"><span class="co">#&gt; loading 'mobster'</span></a>
<a class="sourceLine" id="cb1-12" data-line-number="12"><span class="co">#&gt; Warning: replacing previous import 'ggplot2::last_plot' by</span></a>
<a class="sourceLine" id="cb1-13" data-line-number="13"><span class="co">#&gt; 'plotly::last_plot' when loading 'mobster'</span></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co">#&gt; Warning: replacing previous import 'crayon::style' by 'plotly::style' when</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="co">#&gt; loading 'mobster'</span></a>
<a class="sourceLine" id="cb1-16" data-line-number="16"><span class="co">#&gt; Warning: replacing previous import 'magrittr::extract' by 'tidyr::extract'</span></a>
<a class="sourceLine" id="cb1-17" data-line-number="17"><span class="co">#&gt; when loading 'mobster'</span></a>
<a class="sourceLine" id="cb1-18" data-line-number="18"><span class="co">#&gt;  [ MOBSTER - Model-based clustering in cancer ] </span></a>
<a class="sourceLine" id="cb1-19" data-line-number="19"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="co">#&gt; GitHub :  caravagn/mobster </span></a>
<a class="sourceLine" id="cb1-21" data-line-number="21"><span class="co">#&gt;    WWW :  https://caravagn.github.io/mobster/ </span></a>
<a class="sourceLine" id="cb1-22" data-line-number="22"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-23" data-line-number="23"><span class="co">#&gt;  &gt; MOBSTER is part of the "evoverse" [https://bit.ly/2orn94e] - a collection of packages to implement Cancer Evolution analyses from cancer sequencing data.</span></a>
<a class="sourceLine" id="cb1-24" data-line-number="24"><span class="co">#&gt;  [ VIBER - Variational inference for Binomial mixtures ] </span></a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">#&gt; GitHub :  caravagn/VIBER [https://caravagn.github.io/VIBER/] </span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="co">#&gt;    WWW :  https://caravagn.github.io/VIBER/ </span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-29" data-line-number="29"><span class="co">#&gt;  &gt; VIBER is part of the "evoverse" [https://bit.ly/2orn94e] - a collection of packages to implement Cancer Evolution analyses from cancer sequencing data.</span></a>
<a class="sourceLine" id="cb1-30" data-line-number="30"><span class="co">#&gt; Warning: replacing previous import 'VIBER::plot_mixing_proportions' by</span></a>
<a class="sourceLine" id="cb1-31" data-line-number="31"><span class="co">#&gt; 'mobster::plot_mixing_proportions' when loading 'TINC'</span></a>
<a class="sourceLine" id="cb1-32" data-line-number="32"><span class="co">#&gt; Warning: replacing previous import 'BMix::plot_model_selection' by</span></a>
<a class="sourceLine" id="cb1-33" data-line-number="33"><span class="co">#&gt; 'mobster::plot_model_selection' when loading 'TINC'</span></a>
<a class="sourceLine" id="cb1-34" data-line-number="34"><span class="co">#&gt; Warning: replacing previous import 'VIBER::plot_latent_variables' by</span></a>
<a class="sourceLine" id="cb1-35" data-line-number="35"><span class="co">#&gt; 'mobster::plot_latent_variables' when loading 'TINC'</span></a>
<a class="sourceLine" id="cb1-36" data-line-number="36"><span class="co">#&gt; Warning: replacing previous import 'VIBER::choose_clusters' by</span></a>
<a class="sourceLine" id="cb1-37" data-line-number="37"><span class="co">#&gt; 'mobster::choose_clusters' when loading 'TINC'</span></a>
<a class="sourceLine" id="cb1-38" data-line-number="38"><span class="co">#&gt; Warning: replacing previous import 'BMix::Clusters' by 'mobster::Clusters'</span></a>
<a class="sourceLine" id="cb1-39" data-line-number="39"><span class="co">#&gt; when loading 'TINC'</span></a>
<a class="sourceLine" id="cb1-40" data-line-number="40"><span class="co">#&gt;  [ TINC - Tumour in Normal contamination ] </span></a>
<a class="sourceLine" id="cb1-41" data-line-number="41"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb1-42" data-line-number="42"><span class="co">#&gt; GitHub :  caravagn/TINC </span></a>
<a class="sourceLine" id="cb1-43" data-line-number="43"><span class="co">#&gt;    WWW :  https://caravagn.github.io/TINC/ </span></a>
<a class="sourceLine" id="cb1-44" data-line-number="44"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb1-45" data-line-number="45"><span class="co">#&gt;  &gt; TINC is part of the "evoverse" [https://bit.ly/2orn94e] - a collection of packages to implement Cancer Evolution analyses from cancer sequencing data.</span></a></code></pre></div>
<p>TINC is a simple package that provides only a few functions.</p>
<div id="input-format" class="section level3">
<h3 class="hasAnchor">
<a href="#input-format" class="anchor"></a>Input format</h3>
<p>The input mutation data should report the following information</p>
<pre><code>chr7:96202670:96202670:T:G      33  0   109 47
chr11:127887757:127887757:T:C   26  0   108 52
chr7:147652534:147652534:G:A    37  0   114 10
chr3:18609334:18609334:T:G      31  0   136 15
chr4:86759954:86759954:T:A      29  0   121 14
chr6:95597565:95597565:C:T      39  0   130 70</code></pre>
<p>Columns:</p>
<ol style="list-style-type: decimal">
<li><p>a mutation identifier, which can be anything provided it does not contain duplicates. In this example all mutations are SNVs, and the id contains the chromosome, positions, reference and alternative alleles of the SNV.</p></li>
<li>the number of total reads with both reference/ alternative alleles, in the normal sample.</li>
<li><p>the number of total reads with the alternative allele, in the normal sample.</p></li>
<li>the number of total reads with both reference/ alternative alleles, in the tumour sample.</li>
<li><p>the number of total reads with the alternative allele, in the tumour sample.</p></li>
</ol>
<p>Input (two options):</p>
<ul>
<li>
<em>File</em> : a file that will be loaded by <code>TINC</code>, which should have no column names or headers.</li>
<li>
<em>Dataframe</em> : a data.frame or tibble that has the following columns: <code>id</code>, <code>n_ref_count</code>, <code>n_alt_count</code>, <code>t_ref_count</code>, <code>t_alt_count</code> and <code>filters</code>. . ### Data generator</li>
</ul>
<p>TIN provides a function to generate a random dataset for the analysis, which you can use to play around with TIN.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">sampled_data =<span class="st"> </span><span class="kw"><a href="../reference/random_TIN.html">random_TIN</a></span>(<span class="dt">TIT =</span> <span class="fl">.8</span>, <span class="dt">TIN =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(sampled_data)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; $data</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; # A tibble: 1,000 x 6</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt;    id               n_ref_count n_alt_count t_ref_count t_alt_count filters</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt;    &lt;chr&gt;                  &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt;       &lt;dbl&gt; &lt;chr&gt;  </span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;  1 chr_fake:1:2:A:C          32           0          75          58 PASS   </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt;  2 chr_fake:2:3:A:C          38           0          67          46 PASS   </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt;  3 chr_fake:3:4:A:C          28           0          65          43 PASS   </span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt;  4 chr_fake:4:5:A:C          33           0          68          41 PASS   </span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt;  5 chr_fake:5:6:A:C          25           0          73          42 PASS   </span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt;  6 chr_fake:6:7:A:C          37           1          68          53 PASS   </span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt;  7 chr_fake:7:8:A:C          19           0          77          46 PASS   </span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt;  8 chr_fake:8:9:A:C          24           0          66          50 PASS   </span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt;  9 chr_fake:9:10:Aâ€¦          18           0          64          56 PASS   </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt; 10 chr_fake:10:11:â€¦          27           0          71          46 PASS   </span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; # â€¦ with 990 more rows</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; $plot</span></a></code></pre></div>
<p><img src="TINC_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<p>You can then proceed as follows - using <code>FAST</code> so the analysis is faster</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">TIN_fit =<span class="st"> </span><span class="kw"><a href="../reference/autofit.html">autofit</a></span>(sampled_data<span class="op">$</span>data, <span class="dt">FAST =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="co">#&gt;  [ TINC ~ Load tumour/normal data ] </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co">#&gt;     Input dataframe  1000 x 6 </span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co">#&gt;  Mutations annotated in the normal sample</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt; # A tibble: 1,000 x 11</span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt;    chr   from  to    ref   alt   id    n_ref_count n_alt_count    DP    NV</span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt;  1 chr_â€¦ 1     2     A     C     chr_â€¦          32           0    32     0</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt;  2 chr_â€¦ 2     3     A     C     chr_â€¦          38           0    38     0</span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt;  3 chr_â€¦ 3     4     A     C     chr_â€¦          28           0    28     0</span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt;  4 chr_â€¦ 4     5     A     C     chr_â€¦          33           0    33     0</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt;  5 chr_â€¦ 5     6     A     C     chr_â€¦          25           0    25     0</span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt;  6 chr_â€¦ 6     7     A     C     chr_â€¦          37           1    38     1</span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt;  7 chr_â€¦ 7     8     A     C     chr_â€¦          19           0    19     0</span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt;  8 chr_â€¦ 8     9     A     C     chr_â€¦          24           0    24     0</span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt;  9 chr_â€¦ 9     10    A     C     chr_â€¦          18           0    18     0</span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; 10 chr_â€¦ 10    11    A     C     chr_â€¦          27           0    27     0</span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt; # â€¦ with 990 more rows, and 1 more variable: VAF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt;  Mutations annotated in the tumour sample</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt; # A tibble: 1,000 x 11</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt;    chr   from  to    ref   alt   id    t_ref_count t_alt_count    DP    NV</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;       &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt;  1 chr_â€¦ 1     2     A     C     chr_â€¦          75          58   133    58</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt;  2 chr_â€¦ 2     3     A     C     chr_â€¦          67          46   113    46</span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt;  3 chr_â€¦ 3     4     A     C     chr_â€¦          65          43   108    43</span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt;  4 chr_â€¦ 4     5     A     C     chr_â€¦          68          41   109    41</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt;  5 chr_â€¦ 5     6     A     C     chr_â€¦          73          42   115    42</span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt;  6 chr_â€¦ 6     7     A     C     chr_â€¦          68          53   121    53</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt;  7 chr_â€¦ 7     8     A     C     chr_â€¦          77          46   123    46</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt;  8 chr_â€¦ 8     9     A     C     chr_â€¦          66          50   116    50</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"><span class="co">#&gt;  9 chr_â€¦ 9     10    A     C     chr_â€¦          64          56   120    56</span></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co">#&gt; 10 chr_â€¦ 10    11    A     C     chr_â€¦          71          46   117    46</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="co">#&gt; # â€¦ with 990 more rows, and 1 more variable: VAF &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-41" data-line-number="41"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-42" data-line-number="42"><span class="co">#&gt;  VAF summaries   </span></a>
<a class="sourceLine" id="cb4-43" data-line-number="43"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-44" data-line-number="44"><span class="co">#&gt; # A tibble: 2 x 7</span></a>
<a class="sourceLine" id="cb4-45" data-line-number="45"><span class="co">#&gt;   sample Min.  `1st Qu.` Median Mean  `3rd Qu.` Max. </span></a>
<a class="sourceLine" id="cb4-46" data-line-number="46"><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb4-47" data-line-number="47"><span class="co">#&gt; 1 Normal 0     0         0      0     0         0.04 </span></a>
<a class="sourceLine" id="cb4-48" data-line-number="48"><span class="co">#&gt; 2 Tumour 0.04  0.34      0.38   0.33  0.41      0.92 </span></a>
<a class="sourceLine" id="cb4-49" data-line-number="49"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-50" data-line-number="50"><span class="co">#&gt;      VAF range 0 ~ 0.7  [ c("FALSE 2", "TRUE 998") ] tumour mutations </span></a>
<a class="sourceLine" id="cb4-51" data-line-number="51"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-52" data-line-number="52"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-53" data-line-number="53"><span class="co">#&gt;  MOBSTER fit n = 998</span></a>
<a class="sourceLine" id="cb4-54" data-line-number="54"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-55" data-line-number="55"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-56" data-line-number="56"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-57" data-line-number="57"><span class="co">#&gt;       [AUTOMATIC SETUP]  FAST  - overrides any parameter you have set.</span></a>
<a class="sourceLine" id="cb4-58" data-line-number="58"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-59" data-line-number="59"><span class="co">#&gt;  [ MOBSTER fit ~ N = 998 - random seed 12345 ] </span></a>
<a class="sourceLine" id="cb4-60" data-line-number="60"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-61" data-line-number="61"><span class="co">#&gt;  - Beta(s) K = 1,2 ~ Pareto tail : ON/OFF;</span></a>
<a class="sourceLine" id="cb4-62" data-line-number="62"><span class="co">#&gt;  - Fit by Moments-matching (s = 100, i = random,  Îµ = 1e-06) scoring with reICL</span></a>
<a class="sourceLine" id="cb4-63" data-line-number="63"><span class="co">#&gt;  - Runs 2 x 2 x 2 = 8 without parallel</span></a>
<a class="sourceLine" id="cb4-64" data-line-number="64"><span class="co">#&gt;  - Clusters filter :  Ï€ &gt; 0.02  and  N &gt; 10 </span></a>
<a class="sourceLine" id="cb4-65" data-line-number="65"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-66" data-line-number="66"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-67" data-line-number="67"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-68" data-line-number="68"><span class="co">#&gt; MOBSTER fit completed in 0.06 mins </span></a>
<a class="sourceLine" id="cb4-69" data-line-number="69"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-70" data-line-number="70"><span class="co">#&gt;  BEST FIT: reICL </span></a>
<a class="sourceLine" id="cb4-71" data-line-number="71"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-72" data-line-number="72"><span class="co">#&gt; [ MOBSTER ] N = 998 points with K = 1 Beta clusters, fit by MM in 11 steps [CONVERGED] </span></a>
<a class="sourceLine" id="cb4-73" data-line-number="73"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-74" data-line-number="74"><span class="co">#&gt;   Components (fit)  </span></a>
<a class="sourceLine" id="cb4-75" data-line-number="75"><span class="co">#&gt;      Tail    n = 230 (0.24)   Shape = 1.3 </span></a>
<a class="sourceLine" id="cb4-76" data-line-number="76"><span class="co">#&gt;   Beta C1    n = 768 (0.76)   Mean = 0.39</span></a>
<a class="sourceLine" id="cb4-77" data-line-number="77"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-78" data-line-number="78"><span class="co">#&gt;   Scores (model selection)  </span></a>
<a class="sourceLine" id="cb4-79" data-line-number="79"><span class="co">#&gt;        NLL       BIC       AIC  entropy       ICL reduced.entropy</span></a>
<a class="sourceLine" id="cb4-80" data-line-number="80"><span class="co">#&gt;  -1420.923 -2800.411 -2829.846 47.04639 -2753.365               0</span></a>
<a class="sourceLine" id="cb4-81" data-line-number="81"><span class="co">#&gt;      reICL size</span></a>
<a class="sourceLine" id="cb4-82" data-line-number="82"><span class="co">#&gt;  -2800.411    6</span></a>
<a class="sourceLine" id="cb4-83" data-line-number="83"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-84" data-line-number="84"><span class="co">#&gt;  =-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-85" data-line-number="85"><span class="co">#&gt;   BMix fit n = 768</span></a>
<a class="sourceLine" id="cb4-86" data-line-number="86"><span class="co">#&gt;  =-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-87" data-line-number="87"><span class="co">#&gt;  [ BMix ~ Fitting data ] </span></a>
<a class="sourceLine" id="cb4-88" data-line-number="88"><span class="co">#&gt; Total number of runs:  4 </span></a>
<a class="sourceLine" id="cb4-89" data-line-number="89"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-90" data-line-number="90"><span class="co">#&gt;      Run #  | Samp. # |  Binom. | Beta-Binom. |      Conv. |        ICL</span></a>
<a class="sourceLine" id="cb4-91" data-line-number="91"><span class="co">#&gt; ------------------------------------------------------------------------------</span></a>
<a class="sourceLine" id="cb4-92" data-line-number="92"><span class="co">#&gt;        1/4  |       1 |       1 |           0 |         OK | 1129.60788884076 *</span></a>
<a class="sourceLine" id="cb4-93" data-line-number="93"><span class="co">#&gt;        2/4  |       2 |       1 |           0 |         OK | 843.865767642207 *</span></a>
<a class="sourceLine" id="cb4-94" data-line-number="94"><span class="co">#&gt;        3/4  |       1 |       2 |           0 |         OK | 1059.388</span></a>
<a class="sourceLine" id="cb4-95" data-line-number="95"><span class="co">#&gt;        4/4  |       2 |       2 |           0 |         OK | 1151.393</span></a>
<a class="sourceLine" id="cb4-96" data-line-number="96"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-97" data-line-number="97"><span class="co">#&gt;  *** Best with ICL:  843.8658  </span></a>
<a class="sourceLine" id="cb4-98" data-line-number="98"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-99" data-line-number="99"><span class="co">#&gt; BMix model with K = 1 components: 1 Binomials and 0 Beta-Binomials.</span></a>
<a class="sourceLine" id="cb4-100" data-line-number="100"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-101" data-line-number="101"><span class="co">#&gt; Binomials</span></a>
<a class="sourceLine" id="cb4-102" data-line-number="102"><span class="co">#&gt;       Bin 1 </span></a>
<a class="sourceLine" id="cb4-103" data-line-number="103"><span class="co">#&gt; 0.001664987 </span></a>
<a class="sourceLine" id="cb4-104" data-line-number="104"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-105" data-line-number="105"><span class="co">#&gt; Beta-Binomials</span></a>
<a class="sourceLine" id="cb4-106" data-line-number="106"><span class="co">#&gt; NULL</span></a>
<a class="sourceLine" id="cb4-107" data-line-number="107"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-108" data-line-number="108"><span class="co">#&gt; ICL: 843.8658 </span></a>
<a class="sourceLine" id="cb4-109" data-line-number="109"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-110" data-line-number="110"><span class="co">#&gt;     Binomial peaks 0.00166498707444245 </span></a>
<a class="sourceLine" id="cb4-111" data-line-number="111"><span class="co">#&gt; Mixing proportions 1 </span></a>
<a class="sourceLine" id="cb4-112" data-line-number="112"><span class="co">#&gt;       Clonal score 0.00166498707444245 </span></a>
<a class="sourceLine" id="cb4-113" data-line-number="113"><span class="co">#&gt;                TIN 0.0033299741488849 </span></a>
<a class="sourceLine" id="cb4-114" data-line-number="114"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-115" data-line-number="115"><span class="co">#&gt;  =-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-116" data-line-number="116"><span class="co">#&gt;   VIBER fit n = 1000</span></a>
<a class="sourceLine" id="cb4-117" data-line-number="117"><span class="co">#&gt;  =-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-118" data-line-number="118"><span class="co">#&gt;  [ VIBER - variational fit ] </span></a>
<a class="sourceLine" id="cb4-119" data-line-number="119"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-120" data-line-number="120"><span class="co">#&gt;  INPUT</span></a>
<a class="sourceLine" id="cb4-121" data-line-number="121"><span class="co">#&gt;   Points N = 1000 </span></a>
<a class="sourceLine" id="cb4-122" data-line-number="122"><span class="co">#&gt; Clusters K = 10 (max)</span></a>
<a class="sourceLine" id="cb4-123" data-line-number="123"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-124" data-line-number="124"><span class="co">#&gt; Dirichlet alpha = 1e-06 (conc.)</span></a>
<a class="sourceLine" id="cb4-125" data-line-number="125"><span class="co">#&gt;      Beta a0 = 1; b0 =1 (shape)</span></a>
<a class="sourceLine" id="cb4-126" data-line-number="126"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-127" data-line-number="127"><span class="co">#&gt;      Beta (posterior) prior </span></a>
<a class="sourceLine" id="cb4-128" data-line-number="128"><span class="co">#&gt;  Optimize epsilon = 1e-10; steps =5000; r = 10 </span></a>
<a class="sourceLine" id="cb4-129" data-line-number="129"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-130" data-line-number="130"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-131" data-line-number="131"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-132" data-line-number="132"><span class="co">#&gt; BEST FIT</span></a>
<a class="sourceLine" id="cb4-133" data-line-number="133"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-134" data-line-number="134"><span class="co">#&gt;  [ VIBER - Variational inference for Binomial mixtures ] </span></a>
<a class="sourceLine" id="cb4-135" data-line-number="135"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-136" data-line-number="136"><span class="co">#&gt;     Points N = 1000 </span></a>
<a class="sourceLine" id="cb4-137" data-line-number="137"><span class="co">#&gt; Dimensions W = 2 </span></a>
<a class="sourceLine" id="cb4-138" data-line-number="138"><span class="co">#&gt;        Fit CONVERGED (24 steps; eps. 1e-10) with Variational </span></a>
<a class="sourceLine" id="cb4-139" data-line-number="139"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-140" data-line-number="140"><span class="co">#&gt; Binomial parameters (2 digits rounded) </span></a>
<a class="sourceLine" id="cb4-141" data-line-number="141"><span class="co">#&gt;  # A tibble: 2 x 10</span></a>
<a class="sourceLine" id="cb4-142" data-line-number="142"><span class="co">#&gt;      C5   C10    C6    C1    C2    C3    C4    C7    C8    C9</span></a>
<a class="sourceLine" id="cb4-143" data-line-number="143"><span class="co">#&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-144" data-line-number="144"><span class="co">#&gt; 1  0     0     0.05   0.5   0.5   0.5   0.5   0.5   0.5   0.5</span></a>
<a class="sourceLine" id="cb4-145" data-line-number="145"><span class="co">#&gt; 2  0.39  0.09  0.91   0.5   0.5   0.5   0.5   0.5   0.5   0.5</span></a>
<a class="sourceLine" id="cb4-146" data-line-number="146"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-147" data-line-number="147"><span class="co">#&gt; Proportions (ordered)  </span></a>
<a class="sourceLine" id="cb4-148" data-line-number="148"><span class="co">#&gt; # A tibble: 10 x 2</span></a>
<a class="sourceLine" id="cb4-149" data-line-number="149"><span class="co">#&gt;    entry         value</span></a>
<a class="sourceLine" id="cb4-150" data-line-number="150"><span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb4-151" data-line-number="151"><span class="co">#&gt;  1 C5    0.785        </span></a>
<a class="sourceLine" id="cb4-152" data-line-number="152"><span class="co">#&gt;  2 C10   0.213        </span></a>
<a class="sourceLine" id="cb4-153" data-line-number="153"><span class="co">#&gt;  3 C6    0.00200      </span></a>
<a class="sourceLine" id="cb4-154" data-line-number="154"><span class="co">#&gt;  4 C1    0.00000000100</span></a>
<a class="sourceLine" id="cb4-155" data-line-number="155"><span class="co">#&gt;  5 C2    0.00000000100</span></a>
<a class="sourceLine" id="cb4-156" data-line-number="156"><span class="co">#&gt;  6 C3    0.00000000100</span></a>
<a class="sourceLine" id="cb4-157" data-line-number="157"><span class="co">#&gt;  7 C4    0.00000000100</span></a>
<a class="sourceLine" id="cb4-158" data-line-number="158"><span class="co">#&gt;  8 C7    0.00000000100</span></a>
<a class="sourceLine" id="cb4-159" data-line-number="159"><span class="co">#&gt;  9 C8    0.00000000100</span></a>
<a class="sourceLine" id="cb4-160" data-line-number="160"><span class="co">#&gt; 10 C9    0.00000000100</span></a>
<a class="sourceLine" id="cb4-161" data-line-number="161"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-162" data-line-number="162"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-163" data-line-number="163"><span class="co">#&gt; COMPLETED: 0.07 mins, with status CONVERGED </span></a>
<a class="sourceLine" id="cb4-164" data-line-number="164"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-165" data-line-number="165"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-166" data-line-number="166"><span class="co">#&gt;  Selecting Binomial clusters (F1,2-heuristic).</span></a>
<a class="sourceLine" id="cb4-167" data-line-number="167"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb4-168" data-line-number="168"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-169" data-line-number="169"><span class="co">#&gt; F1.       Cluster size 0.02 </span></a>
<a class="sourceLine" id="cb4-170" data-line-number="170"><span class="co">#&gt; F2. Num. of dimensions 0 </span></a>
<a class="sourceLine" id="cb4-171" data-line-number="171"><span class="co">#&gt; F2. Min. Binomial peak 0 </span></a>
<a class="sourceLine" id="cb4-172" data-line-number="172"><span class="co">#&gt; # A tibble: 10 x 7</span></a>
<a class="sourceLine" id="cb4-173" data-line-number="173"><span class="co">#&gt;    cluster            pi Above_cut F1    F2    accepted new.labels</span></a>
<a class="sourceLine" id="cb4-174" data-line-number="174"><span class="co">#&gt;    &lt;chr&gt;           &lt;dbl&gt;     &lt;dbl&gt; &lt;lgl&gt; &lt;lgl&gt; &lt;lgl&gt;    &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb4-175" data-line-number="175"><span class="co">#&gt;  1 C5      0.785                 2 TRUE  TRUE  TRUE     C1        </span></a>
<a class="sourceLine" id="cb4-176" data-line-number="176"><span class="co">#&gt;  2 C10     0.213                 2 TRUE  TRUE  TRUE     C2        </span></a>
<a class="sourceLine" id="cb4-177" data-line-number="177"><span class="co">#&gt;  3 C1      0.00000000100         2 FALSE TRUE  FALSE    C3        </span></a>
<a class="sourceLine" id="cb4-178" data-line-number="178"><span class="co">#&gt;  4 C2      0.00000000100         2 FALSE TRUE  FALSE    C4        </span></a>
<a class="sourceLine" id="cb4-179" data-line-number="179"><span class="co">#&gt;  5 C3      0.00000000100         2 FALSE TRUE  FALSE    C5        </span></a>
<a class="sourceLine" id="cb4-180" data-line-number="180"><span class="co">#&gt;  6 C4      0.00000000100         2 FALSE TRUE  FALSE    C6        </span></a>
<a class="sourceLine" id="cb4-181" data-line-number="181"><span class="co">#&gt;  7 C6      0.00200               2 FALSE TRUE  FALSE    C7        </span></a>
<a class="sourceLine" id="cb4-182" data-line-number="182"><span class="co">#&gt;  8 C7      0.00000000100         2 FALSE TRUE  FALSE    C8        </span></a>
<a class="sourceLine" id="cb4-183" data-line-number="183"><span class="co">#&gt;  9 C8      0.00000000100         2 FALSE TRUE  FALSE    C9        </span></a>
<a class="sourceLine" id="cb4-184" data-line-number="184"><span class="co">#&gt; 10 C9      0.00000000100         2 FALSE TRUE  FALSE    C10</span></a></code></pre></div>
<p>Summary of the results (with an S3 object)</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(TIN_fit)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt;  [ TINC - Profiler for bulk samples TIN/TIT contamination ] </span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt;  Input :  n = 998 used out of 1000 annotated (100%) </span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt;         TIT :  79%  ~ n = 768 clonal mutations, cluster C1 </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt;         TIN :  0%  ~ n = 38 with VAF &gt; 0 </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt;    QC Tumour   Good purity (65-85%)</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt;    QC Normal   No Contamination &lt;1%</span></a></code></pre></div>
<p>Plot the results (with an S3 object)</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(TIN_fit)</a></code></pre></div>
<p><img src="TINC_files/figure-html/unnamed-chunk-5-1.png" width="960"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Giulio Caravagna.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
